vlmd-submission-tools/vlmd_submission_tools/subcommands/read_and_validate_dictionary.py

```diff
@@ -86,73 +98,119 @@ def main(cls, options: Namespace) -> None:
         local_path = None
-        is_valid_dictionary = None
-        errors_list = None
+        is_valid_dictionary = False
+        validation_error = None
+        errors_list = []
+        report_json = {
+            "json_local_path": None,
+            "is_valid_dictionary": is_valid_dictionary,
+            "errors": errors_list
+        }
 
         # download from url and save local copy
         try:
-            local_path = cls._download_from_url(file_type, dictionary_url, json_local_path)
+            local_path = cls._download_from_url(file_type, dictionary_url, input_dict_path)
             if local_path:
-                logger.info(f"Data dictionary saved in {local_path}")
-
-        except Exception as e:
-            logger.error(f"Could not read dictionary from url {dictionary_url}")
-            logger.error(e)
-            logger.error(f"Exception type = {type(e)}")
+                logger.info(f"Input dictionary saved in {local_path}")
+            else:
+                logger.warning("Input file not downloaded")
+
+        except Exception as err:
+            error_message = f"Could not read dictionary from url {dictionary_url}"
+            logger.error(error_message)
+            logger.error(err)
+
+            report_json["errors"] = [error_message]
+            with open(options.output, 'w', encoding='utf-8') as output_file:
+                json.dump(report_json, output_file, ensure_ascii=False, indent=4)
+            logger.info(f"Validation report saved in {options.output}")
             return
 
-        # get validation report with healdata-utils.validate_vlmd
         logger.info(f"Getting validation report for {local_path}")
-        try:
-            if file_type == 'json':
-                result = validate_vlmd_json(local_path)
-            elif file_type == 'csv' or file_type == 'tsv':
-                result = validate_vlmd_csv(local_path)
-            validation_report = result.get('report')
-
-            is_valid_dictionary = validation_report.get('valid')
-            errors_list = validation_report.get('errors')
-        except Exception as e:
-            logger.error(f"Error in validation: {e}")
 
-        logger.info(f"Valid dictionary = {is_valid_dictionary}")
-        logger.info(f"Errors from validation report = {errors_list}")
+        if file_type == 'json':
+            try:
+                # just validate, don't convert
+                is_valid_dictionary = vlmd_validate(
+                    local_path,
+                    file_type=file_type,
+                    output_type="json",
+                    return_converted_output=False,
+                )
+                converted_json_path = local_path
+
+            except Exception as err:
+                validation_error = str(err)
+                # TODO: remove this parsing after the heal-sdk has trimmed down the error size.
+                validation_error = validation_error.split('\n')[0]
```

Comment:
This line of parsing can be removed after the heal-sdk has trimmed down the error message for jsonschema.ValidationErrors

Reply:
Is that what https://github.com/uc-cdis/heal-platform-sdk/pull/26 was doing?

Reply:
Yes, that was the intention of #26. The PR fixed the verbose errors for CSV files but it did not correctly fix the verbose errors for JSON input.

Reply:
I see, I think we can defer this change to a future PR

Reply:
Sounds good, thank you!